### DISCUSSION TITLE
How to define a route guard for component?

### DISCUSSION THREAD

**Discussion by kongku** [nuxt/nuxt#26472] [Questions]
I would like to define route guards for the component, just like using vue directly, to do some before route  leave or update, how can I achieve this? Using onBeforeRouteLeave() or onBeforeRouteUpdate() issues a warning and the callback function is not called.

**Comment by kongku** [comment]
ok, i get it

**Reply by zx-star** [reply]
How can I use it? I used onbeforerouteleave in the page, and it didn't work
